pkg/registry/service/rest.go

```diff
@@ -277,8 +277,8 @@ var _ = rest.Redirector(&REST{})
 
 // ResourceLocation returns a URL to which one can send traffic for the specified service.
 func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
-	// Allow ID as "svcname" or "svcname:port".
-	svcName, portStr, valid := util.SplitPort(id)
+	// Allow ID as "svcname", "svcname:port", or "svcname:port:scheme".
```

Comment:
what about `scheme://svcname:port`? or `scheme:svcname:port`.

It's super weird to have the scheme at the end, to my eyes at least.

I think there should be a bunch of .md documentation that you should update along with this change.